packages/secure_storage/amplify_secure_storage_dart/lib/src/platforms/amplify_secure_storage_cupertino.dart

```diff
@@ -51,9 +51,9 @@ class AmplifySecureStorageCupertino extends AmplifySecureStorageInterface {
   void write({required String key, required String value}) {
     return using((Arena arena) {
       try {
-        _add(key: key, value: value, arena: arena);
```

Comment:
Note: This change isn't related to the bug fix and has no real impact on behavior.

While reviewing Amplify-iOS code, I noticed they first attempt to update the key, and then add it if the key doesn't exist. We were doing the opposite. The update aligns closer with Amplify-swift and is slightly more efficient for updates, which we would expect to occur more frequently than insertions.